### DISCUSSION TITLE
Is there any documentation around migration from grafonnet-lib to grafonnet.

### DISCUSSION THREAD

**Discussion by unjuli** [grafana/grafonnet#98] [Q&A] [answered]
The new grafonnet library seems to have completely different code structure compared to the deprecated version. Is there any useful documentation around migration across the two? My team has a project which was based on grafonnet-lib and we are looking to migrate to the new version. As of now, it seems like almost everything needs a re-write.

Would be extremely useful to have some guidance around it.
Thanks.

P.S. Was also curious if there are any upcoming CONTRIBUTING guidelines for the project? :)

**Comment by malcolmholmes** [comment, marked as answer]
One of the authors of Jsonnet pointed out to us thst the builder pattern we were using was not performant in Jsonnet, so we moved away from it.

We haven't yet written a migration guide (contributions like that very welcome!!).

While the code structure is different, the dashboard JSON isn't. This means there should be strong parallels between the two. Also, you can easily test your migration. If you run the code before and after your migration, you should get (mostly) the same json.

**Comment by unjuli** [comment]
Thanks!
